theano/gof/op.py

```diff
@@ -784,6 +784,28 @@ class Op(utils.object2, PureOp, CLinkerOp):
     Convenience class to bundle `PureOp` and `CLinkerOp`.
 
     """
+
+    # We add a default get_params() implementation which will try to detect params from the op
+    # if params_type is set to a Wrapper. If not, we raise a MethodNotDefined exception.
+    def get_params(self, node):
+        if hasattr(self, 'params_type'):
+            # If params_type is a Wrapper, we try to extract params from the op.
+            if isinstance(self.params_type, theano.gof.wrapper.Wrapper):
+                wrapper = self.params_type
+                op_has_wrap_attributes = True
+                for field in wrapper.fields:
+                    if not hasattr(self, field):
+                        op_has_wrap_attributes = False
+                        break
+                if op_has_wrap_attributes:
```

Comment:
As `get_params()` has to do the job automatically when Wrapper is used, I think I should better look for params directly into the object (with `hasattr(self, field)`).

So I'll update the code here.